### DISCUSSION TITLE
match-all pattern question

### DISCUSSION THREAD

**Discussion by Solarunit** [sharkdp/fd#1528] [Q&A]
I spent some time figuring out why search doesn't work in specified folder.
I tried:

`fd --extension txt /mnt/dir/` <-- doesn't work
`fd * --extension txt /mnt/dir/` <-- doesn't work

Why you need to use `.`  for matching all pattern? And why not `*`?
this one worked:

`fd . --extension txt /mnt/dir/`

I think it's confusing, because `.` is current folder charecter.

**Comment by tavianator** [comment]
Because `fd` uses regular expressions by default, not globs

**Reply by Solarunit** [reply]
Why `fd --extension txt` works from the current folder? You don't have to specify `.` 
Why not omit `.` when searching in specified folder?

**Reply by tavianator** [reply]
The first argument is always interpreted as a pattern.  So if you write

```console
$ fd --extension txt /mnt/dir
```

then `/mnt/dir` is interpreted as the *pattern*.  If that's not what you want, you have to specify something else as the pattern, or use

```console
$ fd --extension txt --search-path /mnt/dir
```

**Reply by Solarunit** [reply]
I see, for some reason I haven't noticed `--search-path` option, thanks
